docs/source/working_with_aiida/caching.rst

```diff
@@ -4,45 +4,90 @@
+ 2. Run ``verdi -p aiida2 rehash`` in order to compute hashes of all nodes in your existing database
 
-    In [4]: n1.store()
-    Out[4]: u'test string'
+From this point onwards, when you submit a new calculation, AiiDA will compute hashes of the type of calculation and its inputs (see :ref:`caching_matches`). If the hash of the calculation is found to be already present in the database, AiiDA will reuse its results without repeating the actual calculation.
 
-    In [5]: n2 = Str('test string')
+.. _fig_caching:
+.. figure:: include/images/caching.png
+  :align: center
+  :height: 350px
 
-    In [6]: n2.store(use_cache=True)
-    Out[6]: u'test string'
+  When resusing a cached calculation, AiiDA will link up to the input nodes as usual, and make a copy of both the calculation node **C** and its outputs **D3**.
 
-    In [7]: print('UUID of n1:', n1.uuid)
-    UUID of n1: 956109e1-4382-4240-a711-2a4f3b522122
+In order to ensure that the provenance graph with and without caching is the same,
+AiiDA creates a *copy* of the orginial calculation node, linked to the original inputs, as well as a to a copy of the original outputs, as shown in :numref:`fig_caching`.
 
-    In [8]: print('n2 is cached from:', n2.get_extra('_aiida_cached_from'))
-    n2 is cached from: 956109e1-4382-4240-a711-2a4f3b522122
+.. note:: Caching is **not** implemented at the WorkChain/workfunction level (see :ref:`caching_limitations` for details).
+
+
+.. _caching_matches:
 
-As you can see, passing ``use_cache=True`` to the ``store`` method enables using the cache. The fact that ``n2`` was created from ``n1`` is stored in the ``_aiida_cached_from`` extra of ``n2``.
+How are nodes hashed?
+---------------------
+
+By default, the hash of a node is computed from:
+
+* all attributes of the node, except the ``_updatable_attributes``
+* the ``__version__`` of the package which defined the node class
+* the content of the repository folder of the node
+* the UUID of the computer, if the node is associated with one
 
-When running a ``CalcJob`` through the ``Process`` interface, you cannot directly set the ``use_cache`` flag when the calculation node is stored internally. Instead, you can pass the ``_use_cache`` flag to the ``run`` or ``submit`` method.
+Once a node is stored in the database, its hash is stored in the ``_aiida_hash`` extra, and this extra is used to find matching nodes.
+If a node of the same class with the same hash already exists in the database, this is considered a cache match. 
 
-Caching is **not** implemented for workchains and workfunctions. Unlike calculations, they can not only create new data nodes, but also return exsting ones. When copying a cached workchain, it's not clear which node should be returned without actually running the workchain. This is explained in more detail in the section :ref:`caching_provenance`.
+Use the :meth:`.get_hash() <aiida.orm.nodes.Node.get_hash>` method to check the hash of any node.
```

Comment:
the links :meth:`.get_hash() <aiida.orm.nodes.Node.get_hash>` can be written as :meth:`~aiida.orm.nodes.Node.get_hash` i.e. simply a tilde in front to get only the method (or class) name

Reply:
thanks - mentioned by Casper as well.
will fix the occurences (copied from dominik's text)